js/interface.js

```diff
@@ -937,16 +941,22 @@ var DynamicLists = (function() {
 
       loadingPromise = new Promise(function(resolve) {
         if (!_this.config.dataSourceId) {
-          _this.updateFieldsWithColumns(_this.config.defaultColumns);
-          initDataSourceProvider(_this.config.dataSourceId);
+          _this.initDefaultDatasourceProvider();
 
           resolve();
         } else {
-          $document.on('datasource-selected', function() {
+          // Check data source
+          _this.getDataSourceById(_this.config.dataSourceId).then(function() {
+            $document.on('datasource-selected', function() {
+              resolve();
+            });
+
+            _this.initDataSourceProvider(_this.config.dataSourceId);
```

Comment:
It doesn't look like this function is defined in the `_this` scope. Did you test your code?

Rejected +1

Reply:
> It doesn't look like this function is defined in the `_this` scope. Did you test your code?
> 
> Rejected +1

Code is tested and completely work

Reply:
> It doesn't look like this function is defined in the `_this` scope. Did you test your code?
> 
> Rejected +1

I'll fix that